src/zenml/cli/secret.py

```diff
@@ -39,472 +39,483 @@
-    via command line arguments:
-        zenml secret register secret_two --username=admin --password=secret
-    - register a secret with the name `secret_three` interactively and
-    conforming to a schema named `aws` (which is defined in the `aws`
-    integration):
-        zenml integration install aws
-        zenml secret register secret_three -i --schema=aws
-    - register a secret with the name `secret_four` from command line arguments
-    and conforming to a schema named `aws` (which is defined in the `aws`
-    integration). Also load the value for the field `secret_token` from a
-    local file:
-        zenml integration install aws
-        zenml secret register secret_four --schema=aws \
-            --aws_access_key_id=1234567890 \
-            --aws_secret_access_key=abcdefghij \
-            --aws_session_token=@/path/to/token.txt
-
-
-    Args:
-        secrets_manager: The secrets manager to use.
-        name: The name of the secret to register.
-        secret_schema_type: The schema to use for validation.
-        interactive: Whether to use interactive mode to enter the secret values.
-        args: Command line arguments.
-    """
-    # flake8: noqa: C901
-
-    # TODO [ENG-871]: Formatting for `zenml secret register --help` currently
-    #  broken.
-    # TODO [ENG-725]: Allow passing in json/dict when registering a secret as an
-    #   additional option for the user on top of the interactive
-    try:
-        parsed_args = parse_unknown_options(args, expand_args=True)
-    except AssertionError as e:
-        error(str(e))
-
-    if "name" in parsed_args:
-        error("You can't use 'name' as the key for one of your secrets.")
-    elif name == "name":
-        error("Secret names cannot be named 'name'.")
-
-    if name.startswith("--"):
-        error(
-            "Secret names cannot start with '--' The first argument must be."
-            "the secret name."
+        @secrets_manager_group.group(
+            cls=TagGroup,
+            help="Commands for interacting with secrets.",
         )
-
-    if "name" in parsed_args:
-        error("Secret names cannot be passed as arguments.")
-
-    try:
-        from zenml.secret.secret_schema_class_registry import (
-            SecretSchemaClassRegistry,
+        @click.pass_context
+        def secret(ctx: click.Context) -> None:
+            """List and manage your secrets.
+
+            Args:
+                ctx: Click context.
+            """
+            repo = Repository()
+            active_stack = repo.zen_store.get_stack(name=repo.active_stack_name)
+            secrets_manager_wrapper = active_stack.get_component_wrapper(
+                StackComponentType.SECRETS_MANAGER
+            )
+            if secrets_manager_wrapper is None:
+                error(
+                    "No active secrets manager found. Please create a secrets manager "
+                    "first and add it to your stack."
+                )
+                return
```

Comment:
Here's an interesting idea: what if you don't even make these commands visible (don't register them) if the current stack doesn't include a secrets manager ? Would that be too confusing for users ?

Reply:
I think I'd find that confusing, and also possibly would lead to confusion when debugging with a user, perhaps? I like the principle of the idea, but presumably we should also do the same for everything else? I.e. the CLI only shows you commands that are available to you?